lib/travis/build/script/node_js.rb

```diff
@@ -58,6 +71,19 @@ def cache_slug
           super << '--node-' << version
         end
 
+        def setup_cache
+          if data.cache?(:yarn)
+            sh.fold 'cache.yarn' do
+              sh.echo ''
+              directory_cache.add '$HOME/.yarn-cache'
```

Comment:
This has moved to `$HOME/.cache/yarn` and so currently `cache: yarn` does not really work, the cache is empty.

Reply:
This was updated 2 years ago: https://github.com/travis-ci/travis-build/blob/master/lib/travis/build/script/node_js.rb#L95

Reply:
Ugh, :facepalm: Sorry for the noise! 

(That said, I still do have problems with empty yarn cache between build stages, but haven't nailed it down to a self-contained reproducible example, yet)

Reply:
ah, that is a different issue, caches are individual per job. There is not built in facility to share artifacts between jobs and stages right now. We are looking into addressing this.